edi_oca/models/edi_exchange_record.py

```diff
@@ -91,6 +95,11 @@ class EDIExchangeRecord(models.Model):
         comodel_name="edi.exchange.record",
         inverse_name="parent_id",
     )
+    secondary_record_ids = fields.One2many(
+        string="Secondary records",
+        comodel_name="edi.exchange.record.secondary.record",
+        inverse_name="exchange_record_id",
+    )
```

Comment:
```suggestion
    related_record_ids = fields.One2many(
        string="Related records",
        comodel_name="edi.exchange.related.record"
        inverse_name="exchange_record_id",
    )
```
And then:
* always use the relation instead of the direct reference
* make model and res_id computed on the 1st record
* adapt related record methods